### DISCUSSION TITLE
How can i make sure that every PR has a changeset or not

### DISCUSSION THREAD

**Discussion by viveksuarez7** [changesets/changesets#1119] [General]
Hey Everyone , i am using changeset for generating changelog generation and version update . Sometimes some PR's dont have a changeset file and they are pushed to main .We are a team of seven and regularly push pr and do a version update in every 14 days.

I went through the recommended solution i cannot use changeset bot or github actions
The command changeset status --since=main  never breaks beacuse there are changeset generated from other developers are there in our latest branches . Is there any other process i can use to enable my pr's dont miss a changeset ?

**Comment by Andarist** [comment]
`changeset status --since` should cover your use case. I'm not sure how changesets added by other developers affect you here since this command should only check for changesets added **since** your target branch. Unless you are syncing with other branches than that target branch I'm not sure how that would be relevant. A repro case would be helpful to understand what is going on.

**Comment by viveksuarez7** [comment]
main is our root branch where all the pr are merged . When a PR is merged the .changeset file gets added with a changeset.md file . 
The developer takes in the pull from the latest main to develop feature . the branch is named feat/to-do

I am giving you example of the file structure how to will look like of the brancg feat/to-do
.changeset|
          ranndom-changeset.md (Changeset from developer 1)
          random-changeset2.md(Changeset from developer 2)
  src
 package.json 
 
now if do a npm run changeset status --since=main it will not throw an error since the the branch already has couple of changeset.md file

**Comment by Andarist** [comment]
Hm, I took a look into how `status` command is implemented ([here](https://github.com/changesets/changesets/blob/1ea308ab98beeefb5dc6773d277d8a7e08633137/packages/cli/src/commands/status/index.ts#L16)). Packages are tested against the `since` ref and the same is supposed to happen for changeset files, eventually we get to those lines:
https://github.com/changesets/changesets/blob/1ea308ab98beeefb5dc6773d277d8a7e08633137/packages/read/src/index.ts#L37-L43

So... I think that it should work. If you manage to write a failing test case in our repo for this then I could take a look at it, investigate, and provide a fix. I could also just accept a PR (with a test case) if you manage to fix your issue. Without those... I'm not sure how to investigate this on my own. From my PoV, it should just work today.

**Comment by viveksuarez7** [comment]
Hey Andarist , Thanks for your help yess changeset status --since=main resolve the issue ... I had not defined it inside my script tag in package.json and was trying to do an npm run changeset --status . I was able to connect the it to the husky pre-commits aswell. 
Thanks for your great help.
